### PR TITLE
43:Delete choice deletes the choice

### PR DESCRIPTION
**Pull request by valentinbujdoso**
Delete choice deletes the choice

### CODE REVIEW

File: src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+public class ChoiceDeleteValidationTest extends CreatedDefaultVoteRegistry {
+
+	public ChoiceDeleteValidationTest() {
+		super();
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("delete choice")
+	@tested_behaviour("deletes the choice")
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java]: it is @tested_behaviour("validates inputs")
[Deviation in model](http://adadocs.demokracia.rulez.org/PDEngine/edemo/master/index.html#9386718b-3caf-4152-8386-0df9deeb35d3-3044c21c-7606-4098-8176-9dd5b93c20ad-af499e94-b21e-4295-8f22-efe9e9e0be15)

File: src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+		String invalidvoteId = RandomUtils.createRandomKey();
+		String choiceId = voteManager.addChoice(adminInfo.adminKey, adminInfo.voteId, "choice1", "user");
+		assertThrows(
+			() -> voteManager.deleteChoice(invalidvoteId, choiceId ,adminInfo.adminKey)
+		).assertMessageIs(String.format("illegal voteId: %s", invalidvoteId));
+	}
+
+	@tested_feature("Manage votes")
+	@tested_operation("delete choice")
+	@tested_behaviour("deletes the choice")
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java]: same as above

File: src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+		String invalidChoiceId = "InvalidChoiceId";
+		assertThrows(
+				() -> voteManager.deleteChoice(adminInfo.voteId, invalidChoiceId, adminInfo.adminKey)
+			).assertMessageIs(String.format("Illegal choiceId: %s", invalidChoiceId));
+	}
+	
+
+	@tested_feature("Manage votes")
+	@tested_operation("delete choice")
+	@tested_behaviour("deletes the choice")
```
Comment (magwas) [inline src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java]: same as above

**Review by magwas — CHANGES_REQUESTED** [review]
please annotate behaviour correctly

File: src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java (hunk 1 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java (hunk 2 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java]: Fixed

File: src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java (hunk 3 above)
Comment (valentinbujdoso) [reply src/test/java/org/rulez/demokracia/pdengine/ChoiceDeleteValidationTest.java]: Fixed

**Review by magwas — APPROVED** [review]
All okay.
Please do not force-push.  You can add commits fixing things.

**Comment by magwas** [comment]
testcaseCount=4

**Pull request merged**